packages/core/src/stablecoin.ts

```diff
@@ -0,0 +1,369 @@
+import { Contract, ContractBuilder } from './contract';
+import { Access, setAccessControl, requireAccessControl } from './set-access-control';
+import { addPauseFunctions } from './add-pausable';
+import { defineFunctions } from './utils/define-functions';
+import { CommonOptions, withCommonDefaults, defaults as commonDefaults } from './common-options';
+import { setUpgradeable } from './set-upgradeable';
+import { setInfo } from './set-info';
+import { printContract } from './print';
+import { ClockMode, clockModeDefault, setClockMode } from './set-clock-mode';
+
+export interface StablecoinOptions extends CommonOptions {
+  name: string;
+  symbol: string;
+  burnable?: boolean;
+  pausable?: boolean;
+  premint?: string;
+  mintable?: boolean;
+  permit?: boolean;
+  limitations?: boolean | "allowlist" | "blocklist";
```

Comment:
Is there any need to support `true`, or should this be reduced to just be 
```suggestion
  limitations?: false | "allowlist" | "blocklist";
```
This would allow the logic in `addLimitations` to be simplified.